src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -252,36 +258,64 @@ export default class QueryEditorUI extends Component<Props, State> {
     }
   };
 
+  private fetchIndexPatterns = debounce(
```

Comment:
yeah i believe index patterns service might be easier. it's ok can be fast follow